### DISCUSSION TITLE
Expect `withTiming` animations to be executed at render time, but frozen in the result of execution

### DISCUSSION THREAD

**Discussion by sukjae** [software-mansion/react-native-reanimated#4050] [Q&A]
Expect `withTiming` animations to be executed at render time, but frozen in the result of execution.

For example, in the example below, we expected to see an animation with a width of 100 to 200 and a duration of 1000ms at render time.

```jsx
  const style = useAnimatedStyle(() => {
    return {
        width: withTiming(200,{duration:1000})
    };
  });

  return (
    <Animated.View style={[styles.test, style]}>
      <Text
     
      >
        hello Reanimated
      </Text>
    </Animated.View>
  );
}

const styles = StyleSheet.create({
    test: {
      width: 100
    },
  });
```

But it is fixed at 200 at render time.

Is it right way to use something like this for this?

ex1)
```jsx
  const style = useAnimatedStyle(() => {
    return {
        width: withSequence(withTiming(100,{duration:0}), withTiming(200,{duration:1000}))
    };
  });
//...same as above
```
ex1)

```jsx
  const show = useState(false)
  const style = useAnimatedStyle(() => {
    return {
        width:  withTiming(show?200:100,{duration:1000})
    };
  },[show]);
//...same as above
```

Also, 200 is passed as the value of the second argument of `onStart` of the internal logic of withTiming. I wonder what the meaning of the second argument of onStart is. https://github.com/software-mansion/react-native-reanimated/blob/main/src/reanimated2/animation/timing.ts#L69

**Comment by marcocaldera** [comment]
I'm facing the same issue. In older version of reanimated (in production I'm using 2.8) it was working. I suggest we should maybe open an issue

**Reply by piaskowyk** [reply]
Interesting, I think that we have never supported this syntax for animation 🤔 Could you show me the code of your animation?

**Reply by marcocaldera** [reply]
Sure! Before your suggestion the animation was like this:

```typescript
  const loadingListFadeValue = useSharedValue(1);
  const loadingListDisplayValue = useSharedValue(1);

  const onAnimationComplete: AnimationCallback = isFinished => {
    if (isFinished) {
      loadingListDisplayValue.value = 0;
    }
  };

  const loadingListWrapperAnimatedStyles = useAnimatedStyle(() => {
    if (isListLoadedAndMounted) {
      loadingListFadeValue.value = withTiming(
        0,
        {
          duration: 300,
          easing: Easing.bezier(0.32, 0, 0.67, 0)
        },
        runOnJS(onAnimationComplete)
      );
    }

    const opacity = interpolate(loadingListFadeValue.value, [0, 1], [0, 1]);
    const display = loadingListDisplayValue.value === 1 ? 'flex' : 'none';

    return {
      opacity,
      display
    };
  });
```

The one above worked up until `3.0.0-rc.7`

----

After your suggestion:

```typescript
  const loadingListFadeValue = useSharedValue(1);
  const loadingListDisplayValue = useSharedValue(1);

  const onAnimationComplete: AnimationCallback = isFinished => {
    if (isFinished) {
      loadingListDisplayValue.value = 0;
    }
  };

  const loadingListWrapperAnimatedStyles = useAnimatedStyle(() => {
    if (isListLoadedAndMounted) {
      loadingListFadeValue.value = 0;
    }

    return {
      opacity: withTiming(
        loadingListFadeValue.value,
        {
          duration: 300,
          easing: Easing.bezier(0.32, 0, 0.67, 0)
        },
        runOnJS(onAnimationComplete)
      ),
      display: loadingListDisplayValue.value === 1 ? 'flex' : 'none'
    };
  });
```
Now with version`3.0.2` it's working almost all the time. Sometimes after the opacity animation is finished there is a jump/re-render caused, I'm still testing but most probably,  by the `display` property that makes the list underneath flash in/out for a second.

The component tree is a list and (on position absolute) a skeleton loader above it that should fade away and not be displayed once the animation is finished. 

Let me know your opinion and if you need more information!

**Reply by piaskowyk** [reply]
I will check if it is a regression. However, your case looks like you want to use Layout Animation of Entering - https://docs.swmansion.com/react-native-reanimated/docs/api/LayoutAnimations/entryAnimations I think that approach with layout animation could be much easier.

**Reply by marcocaldera** [reply]
Is there a way with a layout animation to decide when to trigger it? I want to animate from a skeleton loader to the actual list. To do that I need to wait for the list to be ready and loaded and only then fade-out the skeleton. 🤔

**Reply by piaskowyk** [reply]
Yes, you can't trigger layout animation manually, but if you want to animate fade-out maybe you can use Exiting Layout Animation? The exiting animation will trigger on loader unmount.

**Reply by marcocaldera** [reply]
What do you mean by "The exiting animation will trigger on loader unmount."? How can I trigger the fade-out on demand? I'm not seeing it in the docs. Thanks!

**Reply by piaskowyk** [reply]
https://user-images.githubusercontent.com/36106620/224662114-a91dea84-8462-4cf7-ba25-0814237b90d8.mov

```tsx
import React, { useState } from 'react';
import Animated, {
  FadeIn,
  FadeOut,
} from 'react-native-reanimated';
import { Button, View } from 'react-native';

export function BasicLayoutAnimation() {
  const [visible, setVisible] = useState(true);

  return (
    <View style={{
      flex: 1,
      alignItems: 'center',
      marginTop: 50,
    }}>
      <Button onPress={() => setVisible(!visible)} title="Create/Remove" />
      {visible && (
        <Animated.View
          entering={FadeIn}
          exiting={FadeOut}
          style={{
            width: 100,
            height: 100,
            backgroundColor: 'blue',
          }}
        />
      )}
    </View>
  );
}
```

**Reply by piaskowyk** [reply]
Is it something what you want to achieve?

**Reply by marcocaldera** [reply]
I text you on Discord (React Native Contributors) to easily share a couple of private videos. I'll then post here the final answer 👍

**Comment by piaskowyk** [comment]
Hey 👋
```tsx
const style = useAnimatedStyle(() => {
  return {
    width: withTiming(200)
  };
});

return (
  <Animated.View style={[{ width: 100 }, style]}>
    <Text>hello Reanimated</Text>
  </Animated.View>
);
```
Unfortunately, this code couldn't work because the animation doesn't know the initial value. Reanimated doesn't know from which value should start the animation. But you can achieve your effect using the shared value, like this:
```tsx
const width = useSharedValue(100)
const style = useAnimatedStyle(() => {
  return {
    width: withTiming(width.value)
  };
});

useEffect(() => {
  width.value = 200
})

return (
  <Animated.View style={[{ width: 100 }, style]}>
    <Text>hello Reanimated</Text>
  </Animated.View>
);
```
